### PR TITLE
Run Windows tests on GitHub Actions

### PR DESCRIPTION
**Pull request by pquentin**
This cherry-picks a few commits from urllib3, fixes Python 3.8 on Windows, moves cleancov.py inside noxfile.py (the previous setup was failing), and uses GitHub Actions to test Python 3.5 to 3.8.

I think GitHub Actions is going to pick up the configuration file, however @njsmith will need to copy the CODECOV_TOKEN secret from https://codecov.io/gh/python-trio/hip/settings to an encrypted variable in the hip repo. Coverage upload wasn't working on AppVeyor and is still not working: I intend to work on it later.

### CODE REVIEW

**Comment by pquentin** [comment]
codecov is lying, src/urllib3/util/connection.py is fully covered, see https://codecov.io/gh/python-trio/hip/pull/153/src/src/urllib3/util/connection.py

File: test/conftest.py [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
 if sys.version_info[:2] < (3, 6):
     collect_ignore_glob = ["async/*.py", "with_dummyserver/async/*.py"]
+
+# The Python 3.8+ default loop on Windows breaks Tornado
+@pytest.fixture(scope="session", autouse=True)
+def configure_windows_event_loop():
+    if sys.version_info >= (3, 8) and platform.system() == "Windows":
+        import asyncio
+
+        asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
```
Comment (njsmith) [inline test/conftest.py]: This is going to be a problem at some point – hip itself will need to support, and be tested on, the default event loop :-/. (Not something for this PR to worry about of course.)

**Comment by njsmith** [comment]
I added the CODECOV_TOKEN

File: test/conftest.py (hunk 1 above)
Comment (pquentin) [reply test/conftest.py]: Thanks, opened https://github.com/python-trio/hip/issues/154 so that we don't forget about it

**Comment by pquentin** [comment]
I used the work-around from https://github.com/theacodes/nox/issues/233#issuecomment-532060445 (and copied stuff from their configuration file) to fully move to GitHub Actions for Windows tests. Now we can really remove AppVeyor. :)

**Comment by njsmith** [comment]
Woohoo! I edited the branch protection to remove appveyor as a required check.

Can you replace `build:` with `Windows:`, so the check names are more meaningful?

**Comment by pquentin** [comment]
@njsmith Thanks, done!

**Comment by njsmith** [comment]
Hmm, I meant replacing the `build:` line with `Windows:`, so that the "job" is named Windows instead of the whole CI run. I guess this works too for now, but my way is more future-proof in case we start moving more runs from travis to github actions :-)

**Comment by pquentin** [comment]
But we don't know what the future will look like :) Maybe we'll have multiple OSes in the same "build", or maybe we'll have different files. I still made the change, and it does look better, thanks!

**Comment by njsmith** [comment]
It turns out that only the "job name" shows up in the branch protection UI:

image

So this is definitely nicer in that respect :-)

**Comment by pquentin** [comment]
Thank you for the review!

**Pull request merged**